File: src/main/java/com/jce/spoj/Factorial.java
package com.jce.spoj;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Factorial {

    public static void main(String... args) throws IOException {
        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));

        int tests = Integer.parseInt(readLine(reader));

        while(tests-- > 0) {
            System.out.println(computeZeros(Integer.parseInt(readLine(reader))));
        }
    }

    public static int computeZeros(int facNum) {
        int fivePow = (int)(Math.log10(facNum)/Math.log10(5));
        int zeros = 0;
        for(int p = 1; p <= fivePow; zeros += facNum/(int)(Math.pow(5, p++)));
        return zeros;
    }

    private static String readLine(BufferedReader reader) throws IOException {
        return reader.readLine();
    }
}

File: src/main/c/factorial.c
#include <math.h>
#include <stdio.h>
#include <stdlib.h>

int main(void) {

    int tests;
    long pow5[] = {5,25,125,625,3125,15625,78125,390625,1953125,9765625,48828125,244140625};

    scanf("%d", &tests);

    while(tests-- > 0) {
        long facNum;
        long zeros = 0;
        long p = 0;

        scanf("%ld", &facNum);

        while(p < 12 && pow5[p] <= facNum) {
            zeros += facNum/pow5[p++];
        }

        printf("%ld\n", zeros);
    }

    return 0;
}